server-plugins/chunter-resources/src/index.ts

```diff
@@ -286,15 +269,37 @@ async function OnThreadMessageDeleted (tx: Tx, control: TriggerControl): Promise
  */
 export async function ChunterTrigger (tx: TxCUD<Doc>, control: TriggerControl): Promise<Tx[]> {
   const res: Tx[] = []
-  res.push(
-    ...(await control.ctx.with('OnThreadMessageCreated', {}, async (ctx) => await OnThreadMessageCreated(tx, control)))
-  )
-  res.push(
-    ...(await control.ctx.with('OnThreadMessageDeleted', {}, async (ctx) => await OnThreadMessageDeleted(tx, control)))
-  )
-  res.push(
-    ...(await control.ctx.with('OnChatMessageCreated', {}, async (ctx) => await OnChatMessageCreated(tx, control)))
-  )
+  const actualTx = TxProcessor.extractTx(tx) as TxCreateDoc<ChatMessage>
+
+  if (
+    actualTx._class === core.class.TxCreateDoc &&
+    control.hierarchy.isDerived(actualTx.objectClass, chunter.class.ThreadMessage)
+  ) {
+    res.push(
+      ...(await control.ctx.with(
+        'OnThreadMessageCreated',
+        {},
+        async (ctx) => await OnThreadMessageCreated(tx, control)
+      ))
+    )
+  }
+  if (control.hierarchy.isDerived(actualTx.objectClass, chunter.class.ThreadMessage)) {
```

Comment:
should add `actualTx._class === core.class.TxRemoveDoc` ?

Reply:
Fixed.